katdal/chunkstore_s3.py

```diff
@@ -172,6 +172,17 @@ def has_chunk(self, array_name, slices, dtype):
             else:
                 return True
 
+    def list_chunk_ids(self, array_name):
+        """See the docstring of :meth:`ChunkStore.list_chunk_ids`."""
+        bucket, prefix = self.split(array_name, 1)
+        paginator = self.client.get_paginator('list_objects')
+        page_iter = paginator.paginate(Bucket=bucket, Prefix=prefix,
+                                       PaginationConfig={'PageSize': 10000})
+        keys = [item['Key'] for page in page_iter for item in page['Contents']]
+        # Strip the array name and .npy extension to get the chunk ID string
+        return [key[len(prefix) + 1:-4] for key in keys]
```

Comment:
As for the Npy store, maybe check that the filename actually conforms to the scheme and it isn't some random junk that got added.

Reply:
Done half-heartedly.